src/server/plugins/reactHtml/index.jsx

```diff
@@ -88,12 +100,60 @@ export function renderModuleScripts({
   return orderedLoadedModules.map((moduleName) => {
     const { integrity, url: src } = moduleMap.modules[moduleName][bundle];
     const { clientCacheRevision } = moduleMap;
-    const additionalAttributes = isDevelopmentEnv ? '' : `integrity="${integrity}"`;
-    const scriptSource = isDevelopmentEnv || !clientCacheRevision ? src : `${src}?clientCacheRevision=${clientCacheRevision}`;
-    return `<script src="${scriptSource}" crossorigin="anonymous" ${additionalAttributes}></script>`;
+    return renderScript({
+      src, integrity, isDevelopmentEnv, clientCacheRevision,
+    });
   }).join(isDevelopmentEnv ? '\n          ' : '');
 }
 
+export function renderExternalFallbacks({
+  clientInitialState,
+  moduleMap,
+  isDevelopmentEnv,
+  isLegacy,
+}) {
+  const loadedModules = clientInitialState.getIn(['holocron', 'loaded'], iSet()).toArray();
+  const requiredFallbacks = loadedModules
```

Comment:
this is a pretty expensive looking reduce in terms of time. are we bound to this by the shape of the data that is in loadedModules or is there potentially a better way of doing this. nested objects with arrays are fun 😄

Reply:
This would only be a handful of modules. loadedModules are modules which are required to render the page, not all the modules included in the module map. even them i'm not sure a module map would be ever be large enough to provide performance concerns. 

> are we bound to this by the shape of the data that is in loadedModules
yes, this could be rewritten to remove that additional reduce on 118 but im not sure that it would add much value.